### DISCUSSION TITLE
Query in Supabase throwing timeout error

### DISCUSSION THREAD

**Discussion by soleuil** [supabase/supabase#22194] [Questions]
I have an endpoint in my project to obtain grades from students and some filters in the UI so the user can select some conditions to filter the data. Everything works just fine except for one specific combination that is getting me a timeout error and logs in Supabase do not give any information. 

Basically I have two filters or conditions: "course" and "unit" and the possibilities that I would have are:
1. "course" === '' && "unit" === '' will fetch all grades without any condition. It works and give around 1000 results without any issue.
2. "course" !== '' && "unit" !== will obtain data from a specific unit and a given course. It works perfectly fine.
3.  "course" !== '' && "unit" === '' will obtain all grades from a given course. It works.
4. Only when "course" === '' && "unit" !== '' it will throw a timeout error. This is a scenario where I would want all grades no matter the course from a specific unit. But this is not working. 

Here are some screenshots that I have:
![Captura de pantalla 2024-03-23 a las 20 10 33](https://github.com/supabase/supabase/assets/61110547/5947aca7-e378-4cf6-847c-3a69c594c4d5)

![1](https://github.com/supabase/supabase/assets/61110547/1d379cf2-e2a7-4505-8a31-d080e2050512)

![2](https://github.com/supabase/supabase/assets/61110547/cfe66c15-22ab-4fc4-a413-21e89a8c15a9)

![3](https://github.com/supabase/supabase/assets/61110547/a359b58a-4550-482e-82c2-3bad0451022a)

I hope someone can give me some direction to solve the issue. Thanks.

**Comment by GaryAustin1** [comment]
Have you tried adding an index to these columns?

**Comment by soleuil** [comment]
I solved the issue changing the approach to the issue because there are not actually columns for "course" and "unit" in this table. Those are foreign keys relationships. Instead of doing that query, I changed it to another table ("notes") and it seems to be much more efficient and fast. Now I don't have any error or issue. 

Thank you!
